proto/minder/v1/minder.proto

```diff
@@ -1199,7 +1207,28 @@ message ListEvaluationResultsRequest{
 message ListEvaluationResultsResponse {
     // status is the list of evaluation results.
     // Results will typically be grouped by profile and project.
-    repeated RuleEvaluationStatus status = 1;
+    reserved 1; // deprecated status
+    reserved "status";
+
+    message RuleEvaluationsList {
+        // rule_evaluation_status is the status of the rules
+        repeated RuleEvaluationStatus evaluation_results = 1;
+    }
+
+    // EntityStatus is a map of entities and their evaluation results.
+    message EntityStatus {
+        // status is the status of the entity
+        map<string, RuleEvaluationsList> status = 1;
+    }
+
+    // EntityList is a list of entities
+    message EntityList {
+        repeated EntityStatus entity = 1;
+    }
+
+    // results is a map of entity types to entities and their evaluation results.
+    // results are grouped by entity, i.e. repository, build_environment, artifact.
+    map<string, EntityList> results = 2;
```

Comment:
Rather than using a map here, I would would use a nested message.  In this case (entity-major), I would do it as:

```proto
message EntityProfileEvaluationResults {
  string profile = 1;
  // Not sure if we want to introduce a new type that avoids including profile_id, entity, etc
  repeated RuleEvaluationStatus results = 2;
}

message EntityEvaluationResults {
  EntityTypeId entity = 1;
  repeated EntityProfileEvaluationResults profiles = 2;
}

message ListEvaluationResultsResponse {
  // Each entity selected by the list request will have _single_ entry in entities which contains results of all evaluations for each profile.
  repeated EntityEvaluationResults entities = 2;
}
```

Another option would be to move common parts of the definition, like rule descriptions, into a `metadata` section, like so:

```proto
message RuleMetadata {
  string rule_type = 1;
  string guidance = 2;
  string description = 3;
}

message ListEvaluationResultsResponse {
  repeated RuleMetadata rule_meta = 3;
  repeated EntityEvaluationResults entities = 2;
}
```

This avoid repeating the guidance and description many times, but still leaves e.g. `last_updated`, `remediation_status`, and `details` in the individual RuleEvaluationStatus messages.

I think it's reasonable to have the frontend go through the `entities` list and filter further by `entities[i].entity.type` if that's what they want, rather than producing a 2-level string map with no type assistance as to what the keys mean.

Reply:
Ah, yeah, I was deliberately waiting for your feedback as I what I did didn't felt right. I like your 1st suggestion as it's simpler and it would work relatively the same for someone already processing the result from `GetProfileStatusByName` too (same response).  Thanks again 👍